qcodes/instrument/parameter.py

```diff
@@ -1349,35 +1381,89 @@ def _update_with(self, *,
         def __call__(self) -> ParamDataType:
             return self.get(get_if_invalid=True)
 
-    def __init__(self, name: str, source: Parameter, *args: Any,
+    def __init__(self, name: str, source: Optional[Parameter], *args: Any,
                  **kwargs: Any):
-        self.source = source
 
-        for ka, param in zip(('unit', 'label', 'snapshot_value'),
-                             ('unit', 'label', '_snapshot_value')):
-            kwargs[ka] = kwargs.get(ka, getattr(self.source, param))
+        self._attr_inherit = {"label": {"fixed": False,
+                                        "default_none_value": name},
```

Comment:
default_none_value means "the value of the attr to be used when the source is None", right? if so, then perhaps a bit more specific name such as "value_when_unbound" or "value_when_without_source" (i think i prefer this so far) or "value_when_none_source" or "source_none_value"?

Reply:
Sure